Model/final_test/datasets.py

```diff
@@ -268,8 +268,68 @@ def load_image_without_mcc(self, r):
     cv2.fillPoly(img, [polygon], (BOARD_FILL_COLOR,) * 3)
     return img
 
+class CubeDataSet(DataSet):
+
+  def get_name(self):
+    return 'Cube'
+
+  def regenerate_meta_data(self):
+    meta_data = []
+    print "Loading and shuffle fn_and_illum[]"
+    ground_truth = np.loadtxt(self.get_directory() + 'cube_gt.txt')
+    ground_truth /=np.linalg.norm(ground_truth, axis=1)[..., np.newaxis]
+
+    illums = ground_truth
+    filenames = sorted(os.listdir(self.get_directory() + 'images'))
+
+    for i in range(len(filenames)):
+      fn = filenames[i]
+      meta_data.append(
+          ImageRecord(
+              dataset = self.get_name(),
+              fn = fn,
+              illum = illums[i],
+              mcc_coord=None,
+              img = None))
+
+    random.shuffle(meta_data)
+
+    if DATA_FRAGMENT != -1:
+      meta_data = meta_data[:DATA_FRAGMENT]
+      print 'Warning: using only first %d images...' % len(meta_data)
+
+    meta_data = slice_list(meta_data, [1] * self.get_folds())
+    self.dump_meta_data(meta_data)
+
+  def load_image(self, fn, mask = 1):
+    file_path = self.get_directory() + 'images/' + str(fn)
+    img = np.array(cv2.imread(file_path, -1), dtype='float32')
+    saturation_level = np.amax(img)-2
+    black_level = 2048
+    img = img - black_level
+    img.clip(0)
+    m = np.zeros((img.shape[0], img.shape[1]))
+    for ch in range(0,2):
```

Comment:
Why only 2 channels, should not it be done on all channels BGR?

Reply:
Sorry, typo..will fix right-away. Thank you very much for spotting it.